recipes/dd-opentracing-cpp/all/conanfile.py

```diff
@@ -42,8 +36,22 @@ def _compilers_minimum_version(self):
             "apple-clang": "7",
         }
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_PLUGIN"] = False
+        tc.variables["BUILD_SHARED"] = self.options.shared
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.variables["BUILD_TESTING"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
     def export_sources(self):
-        self.copy("CMakeLists.txt")
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             self.copy(patch["patch_file"])
```

Comment:
```suggestion
        export_conandata_patches(self)
```

A better idiom, since 1.52.0

Reply:
Does not work see below: Error message is not really helpfull so I will rewertthis change:

```
ERROR: dd-opentracing-cpp/1.3.6: Error in export_sources() method, line 56
	self.copy(patch["patch_file"])
	TypeError: string indices must be integers
```

Reply:
Ah it works but the base_path is needed